src/legacy/core_plugins/timelion/public/vis/timelion_request_handler.ts

```diff
@@ -18,42 +18,56 @@
  */
 
 import _ from 'lodash';
+// @ts-ignore
 import { buildEsQuery, getEsQueryConfig } from '@kbn/es-query';
+// @ts-ignore
 import { timezoneProvider } from 'ui/vis/lib/timezone';
 import { toastNotifications } from 'ui/notify';
 import { i18n } from '@kbn/i18n';
 
-const TimelionRequestHandlerProvider = function (Private, $http, config) {
+const TimelionRequestHandlerProvider = function(Private: any, $http: any, config: any) {
   const timezone = Private(timezoneProvider)();
 
   return {
     name: 'timelion',
-    handler: function ({ timeRange, filters, query, visParams }) {
-
+    handler({
+      timeRange,
+      filters,
+      query,
+      visParams,
+    }: {
+      timeRange: any;
+      filters: any;
+      query: any;
+      visParams: any;
```

Comment:
We have interfaces for all of these somewhere... I think you can use `filters: Filter[]` from the `@kbn/es-query` package, and `Query` interface should be exposed from `src/legacy/core_plugins/data/public` I think.

`VisParams` is in `ui/vis`. And `TimeRange` is in `src/plugins/data/public`.

Reply:
Good point. Added.

Also a new interface for as timelion success response has been added.